pinot-core/src/main/java/org/apache/pinot/core/realtime/converter/stats/RealtimeNoDictionaryColStatistics.java

```diff
@@ -63,12 +68,52 @@ public int getCardinality() {
 
   @Override
   public int getLengthOfShortestElement() {
-    return lengthOfDataType(); // Only fixed length data types supported.
+    FieldSpec.DataType dataType = _blockValSet.getValueType();
+    if (dataType == FieldSpec.DataType.STRING || dataType == FieldSpec.DataType.BYTES) {
```

Comment:
Let us introduce a method in FieldType isFixedWidth()? Will be easier when /if we add new data types.

Reply:
Done